lightning/src/ln/channel.rs

```diff
@@ -978,73 +966,46 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
+		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (included_non_dust_htlcs.len() as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+		let (value_to_self, value_to_remote) = if self.is_outbound() {
 			(value_to_self_msat / 1000 - total_fee as i64, value_to_remote_msat / 1000)
 		} else {
 			(value_to_self_msat / 1000, value_to_remote_msat / 1000 - total_fee as i64)
 		};
 
-		let value_to_a = if local { value_to_self } else { value_to_remote };
-		let value_to_b = if local { value_to_remote } else { value_to_self };
+		let mut value_to_a = if local { value_to_self } else { value_to_remote };
+		let mut value_to_b = if local { value_to_remote } else { value_to_self };
 
 		if value_to_a >= (broadcaster_dust_limit_satoshis as i64) {
 			log_trace!(logger, "   ...including {} output with value {}", if local { "to_local" } else { "to_remote" }, value_to_a);
-			txouts.push((TxOut {
-				script_pubkey: chan_utils::get_revokeable_redeemscript(&keys.revocation_key,
-				                                                       if local { self.counterparty_selected_contest_delay } else { self.holder_selected_contest_delay },
-				                                                       &keys.broadcaster_delayed_payment_key).to_v0_p2wsh(),
-				value: value_to_a as u64
-			}, None));
+		} else {
+			value_to_a = 0;
 		}
 
 		if value_to_b >= (broadcaster_dust_limit_satoshis as i64) {
 			log_trace!(logger, "   ...including {} output with value {}", if local { "to_remote" } else { "to_local" }, value_to_b);
-			let static_payment_pk = if local {
-				self.counterparty_pubkeys.as_ref().unwrap().payment_point
-			} else {
-				self.holder_keys.pubkeys().payment_point
-			}.serialize();
-			txouts.push((TxOut {
-				script_pubkey: Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0)
-				                             .push_slice(&WPubkeyHash::hash(&static_payment_pk)[..])
-				                             .into_script(),
-				value: value_to_b as u64
-			}, None));
-		}
-
-		transaction_utils::sort_outputs(&mut txouts, |a, b| {
-			if let &Some(ref a_htlc) = a {
-				if let &Some(ref b_htlc) = b {
-					a_htlc.0.cltv_expiry.cmp(&b_htlc.0.cltv_expiry)
-						// Note that due to hash collisions, we have to have a fallback comparison
-						// here for fuzztarget mode (otherwise at least chanmon_fail_consistency
-						// may fail)!
-						.then(a_htlc.0.payment_hash.0.cmp(&b_htlc.0.payment_hash.0))
-				// For non-HTLC outputs, if they're copying our SPK we don't really care if we
-				// close the channel due to mismatches - they're doing something dumb:
-				} else { cmp::Ordering::Equal }
-			} else { cmp::Ordering::Equal }
-		});
+		} else {
+			value_to_b = 0;
+		}
+
+		let num_nondust_htlcs = included_non_dust_htlcs.len();
+
+		let channel_parameters =
+			if local { self.channel_transaction_parameters.to_directed_holder_is_broadcaster() }
+			else { self.channel_transaction_parameters.to_directed_counterparty_is_broadcaster() };
```

Comment:
It's a bit ugly that switching on `local` is necessary here. Are there any plans to pass the appropriate directed parameters to the function instead?

Reply:
unsure if this will be an improvement, given that there are a couple of other uses of `local` in the function.  but maybe there's a bigger refactor that I'm not thinking of.